integration/devmode/devmode_test.go

```diff
@@ -0,0 +1,286 @@
+	"github.com/hyperledger/fabric/integration/nwo"
+	"github.com/hyperledger/fabric/integration/nwo/commands"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	"github.com/tedsuo/ifrit"
+	"github.com/tedsuo/ifrit/ginkgomon"
+)
+
+var _ = Describe("Devmode", func() {
+	var (
+		testDir          string
+		client           *docker.Client
+		network          *nwo.Network
+		process          ifrit.Process
+		chaincode        nwo.Chaincode
+		legacyChaincode  nwo.Chaincode
+		chaincodeRunner  *ginkgomon.Runner
+		chaincodeProcess ifrit.Process
+		channelName      string
+	)
+
+	BeforeEach(func() {
+		var err error
+		channelName = "testchannel"
+		testDir, err = ioutil.TempDir("", "devmode")
+		Expect(err).NotTo(HaveOccurred())
+
+		client, err = docker.NewClientFromEnv()
+		Expect(err).NotTo(HaveOccurred())
+
+		network = nwo.New(nwo.DevModeSolo(), testDir, client, StartPort(), components)
+
+		network.TLSEnabled = false
+		// turn on dev mode for all peers
+		network.Peer("Org1", "peer0").DevMode = true
+
+		network.GenerateConfigTree()
+		network.Bootstrap()
+
+		networkRunner := network.NetworkGroupRunner()
+		process = ifrit.Invoke(networkRunner)
+		Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+
+	AfterEach(func() {
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+
+		if chaincodeProcess != nil {
+			chaincodeProcess.Signal(syscall.SIGTERM)
+		}
+
+		if network != nil {
+			network.Cleanup()
+		}
+
+		os.RemoveAll(testDir)
+	})
+
+	It("executes chaincode in dev mode using legacy lifecycle", func() {
+		legacyChaincode = nwo.Chaincode{
+			Name:    "mycc",
+			Version: "0.0",
+			Path:    "github.com/hyperledger/fabric/integration/chaincode/simple/cmd",
+			Ctor:    `{"Args":["init","a","100","b","200"]}`,
+			Policy:  `OR ('Org1MSP.member')`,
+		}
+
+		org1peer0 := network.Peer("Org1", "peer0")
+		orderer := network.Orderer("orderer")
+
+		By("setting up the channel")
+		network.CreateAndJoinChannel(orderer, channelName)
+
+		By("building chaincode")
+		chaincodeExecutePath := components.Build(legacyChaincode.Path)
+
+		By("running the chaincode")
+		legacyChaincodeID := legacyChaincode.Name + ":" + legacyChaincode.Version
+		peerChaincodeAddress := network.PeerAddress(org1peer0, nwo.ChaincodePort)
+		envs := []string{
+			"CORE_PEER_TLS_ENABLED=false",
+			"CORE_CHAINCODE_ID_NAME=" + legacyChaincodeID,
+			"DEVMODE_ENABLED=true",
+		}
+		cmd := exec.Command(chaincodeExecutePath, "-peer.address", peerChaincodeAddress)
+		cmd.Env = append(cmd.Env, envs...)
+		chaincodeRunner = ginkgomon.New(ginkgomon.Config{
+			Name:              "chaincode",
+			Command:           cmd,
+			StartCheckTimeout: 15 * time.Second,
+			StartCheck:        "starting up in devmode...",
+		})
+		chaincodeProcess = ifrit.Invoke(chaincodeRunner)
```

Comment:
Are we missing "Eventually(chaincodeProcess.Ready(), network.EventuallyTimeout).Should(BeClosed())" here?

Reply:
I'll handle this and one nit that Matt mentioned a while ago (moving the network configuration out of the standard_networks).

Reply:
Fixed